components/brave_today/browser/channels_controller.cc

```diff
@@ -54,9 +55,13 @@ void ChannelsController::SetChannelSubscribedPref(PrefService* prefs,
 ChannelsController::ChannelsController(
     PrefService* prefs,
     PublishersController* publishers_controller)
-    : prefs_(prefs), publishers_controller_(publishers_controller) {}
+    : prefs_(prefs), publishers_controller_(publishers_controller) {
+  publishers_controller_->AddObserver(this);
+}
 
-ChannelsController::~ChannelsController() = default;
+ChannelsController::~ChannelsController() {
+  publishers_controller_->RemoveObserver(this);
```

Comment:
nit: Using base::ScopedObservation would be better

Reply:
I always forget it exists :laughing: :+1: